src/transactions/tests_helpers.rs

```diff
@@ -0,0 +1,750 @@
+        (mancpfp_priv, mancpfp),
+        (stakeholders_priv, stakeholders),
+        (cosigners_priv, cosigners),
+    )
+}
+
+// Routine for ""signing"" a transaction
+fn satisfy_transaction_input(
+    secp: &secp256k1::Secp256k1<secp256k1::All>,
+    tx: &mut impl RevaultTransaction,
+    input_index: usize,
+    tx_sighash: &SigHash,
+    xprivs: &Vec<bip32::ExtendedPrivKey>,
+    child_number: Option<bip32::ChildNumber>,
+) -> Result<(), Error> {
+    let derivation_path = bip32::DerivationPath::from(if let Some(cn) = child_number {
+        vec![cn]
+    } else {
+        vec![]
+    });
+
+    for xpriv in xprivs {
+        let sig = secp.sign(
+            &secp256k1::Message::from_slice(&tx_sighash).unwrap(),
+            &xpriv
+                .derive_priv(&secp, &derivation_path)
+                .unwrap()
+                .private_key
+                .key,
+        );
+
+        let xpub = DescriptorPublicKey::XPub(DescriptorXKey {
+            origin: None,
+            xkey: bip32::ExtendedPubKey::from_private(&secp, xpriv),
+            derivation_path: bip32::DerivationPath::from(vec![]),
+            wildcard: if child_number.is_some() {
+                Wildcard::Unhardened
+            } else {
+                Wildcard::None
+            },
+        });
+        let key = if let Some(index) = child_number {
+            xpub.derive(index.into())
+        } else {
+            xpub
+        }
+        .derive_public_key(secp)
+        .unwrap();
+
+        tx.add_signature(input_index, key.key, sig, secp)?;
+    }
+
+    Ok(())
+}
+
+fn desc_san_check<P: MiniscriptKey>(desc: &Descriptor<P>) -> Result<(), ScriptCreationError> {
+    match desc {
+        Descriptor::Wsh(wsh) => wsh.sanity_check().map_err(|e| e.into()),
+        _ => unreachable!(),
+    }
+}
+
+/// Derive transactions for a given deployment configuration, asserting some invariants
+pub fn derive_transactions(
+    n_stk: usize,
+    n_man: usize,
+    csv: u32,
+    deposit_prevout: OutPoint,
+    deposit_value: u64,
+    feebump_prevout: OutPoint,
+    feebump_value: u64,
+    // Outpoint and amount of inputs of a Spend
+    unvault_spends: Vec<(OutPoint, u64)>,
+    secp: &secp256k1::Secp256k1<secp256k1::All>,
+) -> Result<(), Error> {
+    // Let's get the 10th key of each
+    let child_number = bip32::ChildNumber::from(10);
+
+    // Keys, keys, keys everywhere !
+    let (
+        (managers_priv, managers),
+        (_, mancpfp),
+        (stakeholders_priv, stakeholders),
+        (cosigners_priv, cosigners),
+    ) = get_participants_sets(n_stk, n_man, secp);
+
+    let uniq_privs: HashSet<Vec<u8>> = managers_priv
+        .iter()
+        .chain(stakeholders_priv.iter())
+        .chain(cosigners_priv.iter())
+        .cloned()
+        .map(|xpriv| xpriv.private_key.to_bytes())
+        .collect();
+    if uniq_privs.len() != managers_priv.len() + stakeholders_priv.len() + cosigners_priv.len() {
+        return Ok(());
+    }
```

Comment:
Why are we returning Ok here?

Reply:
Because i didn't have a `revault_tx::Error` that fitted i guess (and wasn't going to create one just for the test)

Reply:
Can you add a small comment then please? :)

Reply:
Removed it, checked and it's not needed now that i fixed the `cfg(fuzzing)` thing.